addons/payment_demo/static/src/js/express_checkout_form.js

```diff
@@ -0,0 +1,73 @@
+/** @odoo-module */
+
+import { paymentExpressCheckoutForm } from '@payment/js/express_checkout_form';
+import { paymentDemo } from '@payment_demo/js/payment_demo_mixin';
+
+paymentExpressCheckoutForm.include({
+
+    /**
+     * Simulate a feedback from a payment provider and redirect the customer to the status page.
+     *
+     * @override method from payment.express_checkout_form
+     * @private
+     * @param {object} providerData
+     * @return {Promise}
+     */
+    async _prepareExpressCheckoutForm(providerData) {
+        if (providerData.providerCode !== 'demo') {
+            return this._super(...arguments);
+        } 
+    },
+
+    _onClickPay: function (ev) { 
+        this._super(...arguments);
+        let expressShippingAddress = {};
```

Comment:
Since this function is getting overwritten in places, we need to call super in case we're not in demo.

Reply:
Exactly my point. Where do you check we are with the demo provider to ensure what's done here is only done for the demo provider? If this code should always run, whatever the provider, then it could be on `payment`.

Reply:
Hm, since it;s doing nothing maybe there is no point of this code.